app/cdap/components/WrangleHome/Components/OngoingDataExplorations/index.tsx

```diff
@@ -47,80 +56,91 @@ export default function OngoingDataExplorations() {
-          workspace.headers.forEach((element) => {
-            const general = workspace.summary.statistics[element].general;
-            const { empty: empty = 0, 'non-null': nonEmpty = 100 } = general;
-            const nonNull = Math.floor((nonEmpty - empty) * 10) / 10;
-            dataQuality = dataQuality + nonNull;
+        if (responses && Array.isArray(responses) && responses.length) {
+          responses.forEach((workspace, index) => {
+            let dataQuality = 0;
+            workspace.headers.forEach((element) => {
+              const general = workspace.summary.statistics[element].general;
+              const { empty: empty = 0, 'non-null': nonEmpty = 100 } = general;
+              const nonNull = Math.floor((nonEmpty - empty) * 10) / 10;
+
+              dataQuality = dataQuality + nonNull;
+            });
+            const totalDataQuality = dataQuality / workspace.headers.length;
+            expData[index].dataQuality = totalDataQuality;
+            expData[index].count = workspace.count;
+            const final = generateDataForExplorationCard(expData, cardCount);
+
+            /**
+             * if we have setData, then we should send data to parent element as the exploration state is then maintained in parent as well for showing or hiding the title of the parent component
+             */
+            if (setShowExplorations) {
+              setShowExplorations(final && Array.isArray(final) && final.length ? true : false);
+            }
+            setFinalArray(final);
+
+            setLoading(false);
           });
-          const totalDataQuality = dataQuality / workspace.headers.length;
-          setOngoingExpDatas((current) => [
-            ...current.slice(0, index),
-            {
-              ...current[index],
-              dataQuality: totalDataQuality,
-            },
-            ...current.slice(index + 1),
-          ]);
-        });
+        }
+        setLoading(false);
       });
-  };
+  }, []);
 
   useEffect(() => {
     getOngoingData();
   }, []);
 
-  useEffect(() => {
-    const final = generateDataForExplorationCard(ongoingExpDatas);
-    setFinalArray(final);
-  }, [ongoingExpDatas]);
-
   return (
     <Box data-testid="ongoing-data-explore-parent">
-      {finalArray.map((item, index) => {
-        return (
-          <Link
-            to={{
-              pathname: `/ns/${getCurrentNamespace()}/wrangler-grid/${`${item[4].workspaceId}`}`,
-              state: {
-                from: T.translate('features.Breadcrumb.labels.wrangleHome'),
-                path: T.translate('features.Breadcrumb.params.wrangleHome'),
-              },
-            }}
-            style={{ textDecoration: 'none' }}
-          >
-            {index <= 1 && <OngoingDataExplorationsCard item={item} key={index} />}
-          </Link>
-        );
-      })}
+      {finalArray && Array.isArray(finalArray) && finalArray.length ? (
+        finalArray.map((item, index) => {
```

Comment:
instead of finalArray, provide some meaningful name like onGoingDataExplorationsData.

Reply:
Done